server/sonar-server/src/main/java/org/sonar/server/util/cache/DiskCacheById.java

```diff
@@ -0,0 +1,96 @@
+/*
+ * SonarQube, open source software quality management tool.
+ * Copyright (C) 2008-2014 SonarSource
+ * mailto:contact AT sonarsource DOT com
+ *
+ * SonarQube is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * SonarQube is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+package org.sonar.server.util.cache;
+
+import org.apache.commons.io.FileUtils;
+import org.sonar.api.utils.System2;
+import org.sonar.server.util.CloseableIterator;
+import org.sonar.server.util.EmptyIterator;
+import org.sonar.server.util.ObjectInputStreamIterator;
+
+import java.io.File;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.ObjectOutputStream;
+import java.io.Serializable;
+
+/**
+ * Serialize and deserialize objects on disk in different files identified by an id. No search capabilities, only traversal (full scan).
+ */
+public class DiskCacheById<O extends Serializable> {
```

Comment:
I have the feeling that DiskCache and DiskCacheById will have to be replaced later in the sprint. Let's keep this implementation for now.